rook-ceph ansible improvements

vigneshv-ocl opened the pull request:
- remove delegate, localhost
  - ensure to do a check if the subsequent steps runs in the control plane node
  - target the playbook on node1
- replace shell/command tasks with kubernetes tasks
- delete the unnecessary files
- prefix collection for module name

shimpa1 committed:
new branch

shimpa1 committed:
second version, integrated into kubespray

shimpa1 committed:
README

shimpa1 committed:
minor fixes

shimpa1 committed:
fixed typos in README

vigneshv-ocl committed:
rook-ceph ansible improvements

vigneshv-ocl committed:
rook-ceph ansible improvements

vigneshv-ocl committed:
add python3-kubernetes

shimpa1 commented:
I've tested this scenario and it produces errors.
I think we should merge the rook-ceph-1 commit which works, we can deal with the cosmetics later.

vigneshv-ocl commented:
> I've tested this scenario and it produces errors. I think we should merge the rook-ceph-1 commit which works, we can deal with the cosmetics later.
it was a minor interpreter error which was fixed.

Commit:
refactor: unify config, update charts, improve GPU + journalctl handling, support local + CSI use

- Unified provider `attributes` and pricing config across defaults, templates, and host_vars
- Updated Helm chart versions:
  - Nvidia Device Plugin: 0.17.0 → 0.17.1
  - Ingress NGINX: 4.12.0 → 4.12.1
  - Rook-Ceph: 1.16.2 → 1.16.6
- Added Helm v3.17.2 installer in provider prereqs
- Enabled running playbooks locally by copying Helm values to remote host
- Added support for custom Rook CSI `kubeletDirPath`:
  - Templated `rook-ceph-operator.values.yml.j2`
  - Integrated with cluster config script
  - Output hints for setting per-node overrides
- Improved GPU role:
  - Avoid double-installing if toolkit/runtime already present
  - Skip driver reinstall when version matches
  - Removed autoremove and unnecessary module reload logic
  - Prevent backup overwrite of container runtime config
- Refined `journal_cleanup`:
  - Conditionally patch `logrotate` config if initial run passes
  - Skip redundant rotate attempt if no error
- Minor cleanup:
  - Dropped Snap purge/removal step (was redundant or over-aggressive)
  - Added `meta/main.yml` to all roles with author and platform info

andy108369 reviewed (approved):
tested while re-building rtx4090 evergreen provider (consisting of one node).
pushed the fixes.

Commit:
add ansible.cfg

vigneshv-ocl commented on ansible.cfg:
```diff
@@ -0,0 +1,23 @@
+[ssh_connection]
```
why is this needed

andy108369 replied:
it's good to use this as a default config for ansible.
is probably the most tested & vetted config in the world https://github.com/kubernetes-sigs/kubespray/blob/master/ansible.cfg
I am always using it.

Commit:
remove price_* bits from host_vars/node1.yml (per Vignesh)

Commit:
Merge remote-tracking branch 'origin/rook-ceph-1' into rook-ceph-vignesh-patch-v1

andy108369 commented on host_vars/node1.yml:
```diff
@@ -12,11 +12,28 @@ domain: ""                # Domain name
+price_hd_pers_nvme: 0.1
+price_endpoint: 0.05
+price_ip: 5
+price_gpu_mappings: "a100=569,*=569"
```
fixed.

vigneshv-ocl committed:
merge the contents of rook-ceph-defaults.yml to default main.yml

andy108369 commented on roles/rook-ceph/defaults/main.yaml:
```diff
@@ -1,2 +1,16 @@
 rook_ceph_namespace: rook-ceph
 rook_ceph_version: "1.16.6"
+
+ceph_cluster:
```
this gets generated by `./roles/rook-ceph/generate-rook-ceph-cluster-config.sh`
I am not sure we want it in the default section.
